pkg/operator/ceph/cluster/osd/volumes.go

```diff
@@ -86,7 +87,31 @@ func getDeviceMapperVolume() (v1.Volume, v1.VolumeMount) {
 	return volume, volumeMounts
 }
 
-func getPVCOSDVolumes(osdProps *osdProperties) []v1.Volume {
+func getDataBridgeVolumeSource(claimName, configDir, namespace string, inProvisioning bool) v1.VolumeSource {
+	var source v1.VolumeSource
+	if inProvisioning {
+		source.EmptyDir = &v1.EmptyDirVolumeSource{
+			Medium: "Memory",
+		}
+	} else {
+		// We need to use hostPath to prevent multiple OSD pods from launching the same OSD and causing corruption.
+		// Ceph avoids this problem by locking fsid file and block device file under the data bridge volume directory.
+		// These locks are released by kernel once the process is gone, so until the ceph-osd daemon alives, the other
+		// pods (same OSD) will not be able to acquire them and will continue to be restarted.
+		// If we use emptyDir, this exclusive control doesn't work because the lock files aren't shared between OSD pods.
+		hostPathType := v1.HostPathDirectoryOrCreate
+		source.HostPath = &v1.HostPathVolumeSource{
+			Path: filepath.Join(
+				configDir,
```

Comment:
It looks like the `configDir` will be the `dataDirHostPath`. If there are multiple OSDs on the same node, they would all bindmount the same path. Just to make sure the OSDs stay independent, is it possible bindmount subdir specific to the OSD ID?

Reply:
No, each OSD bindmounts its own directory, `<configDir>/<namespace>/<claimName>` than the same `<configDir>`.

Reply:
Oh of course I see the path join now. 

What about the non-pvc question? Does this lock also work for the non-pvc case? I don't see where these changes apply to the non-pvc case.

Reply:
In non-pvc case, I should change the volumeSource of "activate-osd" volume from `emptyDir` to `hostPath` and should find a proper unique name.

Reply:
I fixed to consider non-PVC case too.